internal/pkg/cli/validate.go

```diff
@@ -14,16 +14,50 @@ import (
 )
 
 var (
-	errValueEmpty        = errors.New("value must not be empty")
-	errValueTooLong      = errors.New("value must not exceed 255 characters")
-	errValueBadFormat    = errors.New("value must start with a letter and contain only lower-case letters, numbers, and hyphens")
-	errValueNotAString   = errors.New("value must be a string")
-	errInvalidGitHubRepo = errors.New("value must be a valid GitHub repository, e.g. https://github.com/myCompany/myRepo")
-	errPortInvalid       = errors.New("value must be in range 1-65535")
+	errValueEmpty          = errors.New("value must not be empty")
+	errValueTooLong        = errors.New("value must not exceed 255 characters")
+	errValueBadFormat      = errors.New("value must start with a letter and contain only lower-case letters, numbers, and hyphens")
+	errValueNotAString     = errors.New("value must be a string")
+	errInvalidGitHubRepo   = errors.New("value must be a valid GitHub repository, e.g. https://github.com/myCompany/myRepo")
+	errPortInvalid         = errors.New("value must be in range 1-65535")
+	errS3ValueBadSize      = errors.New("value must be between 3 and 63 characters in length")
+	errS3ValueBadFormat    = errors.New("value must not contain consecutive periods or dashes, or be formatted as IP address")
+	errS3ValueTrailingDash = errors.New("value must not have trailing -")
+	errS3ValueBadCharacter = errors.New("value must contain only alphanumeric characters and .-")
+	errDDBValueBadSize     = errors.New("value must be between 3 and 255 characters in length")
+	errDDBValueBadFormat   = errors.New("value must contain only alphanumeric characters and ._-")
 )
 
+var fmtErrInvalidStorageType = "invalid storage type %s: must be one of %s"
+
 var githubRepoExp = regexp.MustCompile(`(https:\/\/github\.com\/|)(?P<owner>.+)\/(?P<repo>.+)`)
 
+// matches alphanumeric, ._-, from 3 to 255 characters long
+// https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html
+var ddbRegExp = regexp.MustCompile(`^[a-zA-Z0-9\-\.\_]+$`)
+
+// matches alphanumeric, .- from 3 to 63 characters long.
+// https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-s3-bucket-naming-requirements.html
+var s3RegExp = regexp.MustCompile(
```

Comment:
would you mind moving these s3 regexes to a single group?
```go
var (
    s3RegExp =
    s3DashesRegExp =
    s3TrailingDashRegExp = 
    s3IpAddressRegexp =
)
```

Reply:
Done.